helevents/models.py

```diff
@@ -1,6 +1,128 @@
+
+    A mixin class that provides permission check methods
+    for user models.
+    """
+
+    @property
+    def is_external(self):
+        """Check if the user is an external user"""
+        return (
+            not self.organization_memberships.exists()
+            and not self.admin_organizations.exists()
+        )
+
+    def is_admin_of(self, publisher):
+        """Check if current user is an admin user of the publisher organization"""
+        raise NotImplementedError()
+
+    def is_regular_user_of(self, publisher):
+        """Check if current user is a regular user of the publisher organization"""
+        raise NotImplementedError()
+
+    @property
+    def admin_organizations(self):
+        """
+        Get a queryset of organizations that the user is an admin of.
+
+        Is replaced by django_orghierarchy.models.Organization's
+        admin_organizations relation unless implemented in a subclass.
+        """
+        raise NotImplementedError()
+
+    @property
+    def organization_memberships(self):
+        """
+        Get a queryset of organizations that the user is a member of.
+
+        Is replaced by django_orghierarchy.models.Organization's
+        organization_memberships relation unless implemented
+        in a subclass.
+        """
+        raise NotImplementedError()
+
+    def can_create_event(self, publisher, publication_status):
+        """Check if current user can create an event with the given publisher and publication_status"""
+        if self.is_admin_of(publisher):
+            return True
+        # Non-admins can only create drafts.
+        return publication_status == PublicationStatus.DRAFT
+
+    def can_edit_event(self, publisher, publication_status, created_by=None):
+        """Check if current user can edit an event with the given publisher and publication_status"""
+        if self.is_admin_of(publisher):
+            return True
+
+        # Non-admins can only edit drafts.
+        if publication_status == PublicationStatus.DRAFT:
+            if (
+                publisher is None
+                or publisher.id == settings.USER_DEFAULT_ORGANIZATION_ID
+            ):
+                # External users can only edit their own drafts from the default organization.
+                return self.is_external and created_by == self
+            else:
+                # Regular users can edit drafts from organizations they are members of.
+                return self.is_regular_user_of(publisher)
+
+        return False
+
+    def get_editable_events(self, queryset):
+        """Get editable events queryset from given queryset for current user"""
+        if self.is_external:
+            # External users can only edit their own drafts from the default organization.
+            return queryset.filter(
+                created_by=self,
+                publisher__id=settings.USER_DEFAULT_ORGANIZATION_ID,
+                publication_status=PublicationStatus.DRAFT,
+            )
```

Comment:
I'd add `settings.ENABLE_EXTERNAL_USER_EVENT_CREATE`

```suggestion
        if self.is_external:
        
            if not settings.ENABLE_EXTERNAL_USER_EVENT_CREATE:
                return queryset.none()
                
            # External users can only edit their own drafts from the default organization.
            return queryset.filter(
                created_by=self,
                publisher__id=settings.USER_DEFAULT_ORGANIZATION_ID,
                publication_status=PublicationStatus.DRAFT,
            )
```